src/imitation/rewards/discrim_net.py

```diff
@@ -228,17 +289,20 @@ def build_graph(self):
 
     # Construct discriminator logits: $f_{\theta}(s,a)$, predicted rewards,
     # and $\log \pi(a \mid s)$, generator-policy log action probabilities.
-    self._presoftmax_disc_logits = tf.stack(
+    _presoftmax_disc_logits = tf.stack(
         [self.reward_net.reward_output_train, self.log_policy_act_prob_ph],
         axis=1, name="presoftmax_discriminator_logits")  # (None, 2)
 
     # Construct discriminator loss.
     self._disc_loss = tf.nn.sparse_softmax_cross_entropy_with_logits(
         labels=self.labels_ph,
-        logits=self._presoftmax_disc_logits,
-        name="discrim_loss",
+        logits=_presoftmax_disc_logits,
+        name="disc_loss",
     )  # (None,)
 
+    softmax_disc_logits = tf.nn.log_softmax(_presoftmax_disc_logits)
+    self._disc_logits = softmax_disc_logits[:, 0] - softmax_disc_logits[:, 1]
+
```

Comment:
Quick sanity check: It's necessary to do a `log_softmax` before assigning `self._disc_logits` because we need the normalizing constant, correct?

Reply:
`log_softmax` shouldn't change the result, but it is unnecessary because in the next line the normaliser cancels out. Specifically we have:

```
norm_const = sum(exp(_presoftmax_disc_logits), axis=1)

softmax(_presoftmax_disc_logits) = [exp(_presoftmax_disc_logits[0]) / norm_const, exp(_presoftmax_disc_logits[1]) / norm_const]

softmax_disc_logits = log(softmax(_presoftmax_disc_logits))
                    = stack([_presoftmax_disc_logits[:, 0] - log(norm_const), _presoftmax_disc_logits[:, 1] - log(norm_const)], axis=1)

_disc_logits = softmax_disc_logits[:, 0] - softmax_disc_logits[:, 1]
             = _presoftmax_disc_logits[:, 0] - log(norm_const) - (_presoftmax_disc_logits[:, 1] - log(norm_const))
             = presoftmax_disc_logits[:, 0] - _presoftmax_disc_logits[:, 1]
```

Reply:
Thanks for writing this out, makes sense